### DISCUSSION TITLE
ImportError: cannot import name 'ROOT_KEY' from 'pydantic.main'

### DISCUSSION THREAD

**Discussion by PlatinumDiors** [pydantic/pydantic#9130] [Question]
my env info followed:
             pydantic version: 2.6.4
        pydantic-core version: 2.16.3
          pydantic-core build: profile=release pgo=true
                 install path: /Users/gaohongyang/Downloads/project/python/ollama/.venv/lib/python3.12/site-packages/pydantic
               python version: 3.12.0 (v3.12.0:0fb18b02c8, Oct  2 2023, 09:45:56) [Clang 13.0.0 (clang-1300.0.29.30)]
                     platform: macOS-14.1.2-arm64-arm-64bit
             related packages: typing_extensions-4.10.0
                       commit: unknown,

when i run my code, I faced this error "ImportError: cannot import name 'ROOT_KEY' from 'pydantic.main' ", could anyone faced the same issue before? and how to fixed the issue? thx

**Comment by alonsosilvaallende** [comment]
I was encountering the same issue and I downgraded Python from 3.12 to 3.11 and the issue dissapeared.

**Reply by mksaraf** [reply]
[ImportError: cannot import name 'OpenAIEmbeddings' from 'langchain.embeddings.openai' (/Users/manishkumarsaraf/Library/Python/3.9/lib/python/site-packages/langchain/embeddings/openai.py)
](url)

**Comment by Viicos** [comment]
`ROOT_KEY` is a v1 only thing. Importing it from `pydantic.main` in v2 will not work as expected
